dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -2112,27 +2114,410 @@ than a dMY_CTX in another source file.
+=item C<å> and C<Å>
+
+These are like C<_r> and C<_R> respectively, except you can get Perl to
+B<automatically> change your calls of the base function to instead call
+the thread-safe C<_r> version.  The little circle above the letter
+represents a halo (which in European culture represents purity)
+representing that the substituted version is fully thread-safe (on most
+platforms anyway).
+
+These symbols indicate that you can generally ignore thread safety
+issues for these, as your code will work fine on platforms that have
+them, and as good as can be expected on platforms that don't.  (Likely
+those other platforms will be minority platforms with few users, and you
+can just say you don't support threaded operation on them.)
+
+To get this automatic substitution, add this line to your code:
 
  #define PERL_REENTRANT
 
-in your code to enable the wrapping.  It is advisable to do so if you
-are using such functions, as intermixing the C<_r>-forms (as Perl compiled
-for multithreaded operation will do) and the C<_r>-less forms is neither
-well-defined (inconsistent results, data corruption, or even crashes
-become more likely), nor is it very portable.  Unfortunately, not all
-systems have all the C<_r> forms, but using this C<#define> gives you
-whatever protection that Perl is aware is available on each system.
+This must follow your
+
+ #include <perl.h>
+
+line.  If you don't include F<perl.h>, you must instead place this line
+
+ #include <regen/reentr.pl>
+
+before your C<PERL_REENTRANT> definition.
+
+(This facility is available outside the perl core and its extensions
+starting in Perl 5.28.)
+
+In some of these, the base function is thread-safe if called with a
+non-NULL parameter, and the only purpose of the C<_r> version is to
+fail if the call fails to use a safe form.
+
+=item C<a> and C<A>
+
+These are like C<å> and C<Å> respectively, except the lack of a halo
+indicates that the substitued version isn't fully thread-safe.  If an
+C<L</M>> flag also appears, wrap your calls to this function with the
+Perl-furnished locking macros to get full thread-safety.  Otherwise,
+you'll have to examine the man pages.
+
+=item C<B>
+
+This function returns data in a static buffer which may be destroyed by
+the next call in the same thread, or on some platforms, it may be
+destroyed at any time by another thread executing this function or a
+related one.
+
+Some platforms may allocate the buffer per-thread, so it is thread-safe,
+but you have to use the value (or copy it) before the next call.
+
+There may be a re-entrant function available in which the caller
+furnishes the buffer, and which is thread-safe.  Other flags on the
+entry should indicate this.
+
+=item E
+
+This function accesses the program's environment.
+
+Writing to the environment can create a race with other threads either
+reading or writing the environment at the same time.  If all writers (as
+the core does) protect their accesses by surrounding them with
+
+ ENV_LOCK
```

Comment:
All the `*_LOCK`/`*_UNLOCK` lines should have a terminating `;` since that's how they will appear in code.